app/experimenter/docs/adrs/0002-use-react-bootstrap-cra-gql-for-nimbus.md

```diff
@@ -0,0 +1,113 @@
+# Use React, Bootstrap, CRA, and GQL for the Nimbus Front-End Stack
+
+* Status: proposed
+* Deciders: Lauren Zugai, Jody Heavener, [TBD]
+* Date: 2020-09-08
+
+## Context and Problem Statement
+
+The front-end for Experimenter is currently composed of two applications. A more mature application for Normandy experiments, found at `app/experimenter/static/core`, is composed of a mix of Django templates, vanilla JS, jQuery, and React components. A newer React application for Nimbus experiments (previously named Rapid experiments) can be found at `app/experimenter/static/rapid`.
+
+With the decision made to build the Nimbus console as part of the Experimenter codebase ([see ADR #0001](https://github.com/mozilla/experimenter/blob/main/app/experimenter/docs/adrs/0001-build-nimbus-console-in-experimenter.md)), it was determined we likely want a "new front-end." This ADR, while slightly atypical as it does not serve to document a single architectural change, outlines a proposed directory tree change regarding the front-end, a proposed new front-end stack, how it compares to the current `rapid` stack, and how we will benefit from these changes.
+
+## Decision Drivers
+
+Drivers for directory tree changes:
+* A more obvious separation of concerns for:
+  * the front-end and back-end
+  * legacy front-end code and Nimbus front-end code
+
+Drivers for the proposed new stack:
+* Better development tooling
+* A more cohesive fit into the FxA team's stack and processes
+* Development speed
+
+Drivers for starting a new application:
+* Time needed and ease of refactoring the current `rapid` React application vs time needed to start with a new application
+
+## Proposed Changes
+
+### App Directory Tree
+
+Currently, a `static/` directory contains the front-end applications with some shared assets, and a `templates/` directory lives adjacent to it.  To provide a more obvious separation of concerns in the directory tree:
+
+* Rename `static/` to `legacy-ui/` since `static/rapid/` will become obsolete and `static/core/` will be the legacy Normandy platform
+* Move `templates/` into `legacy-ui/`
+* Use `create-react-app` to create a new React application in `app/experimenter/nimbus-ui/`
```

Comment:
Is there any chance of moving this up under `/app` so it's not so heavily nested? This could also help make the distinction between front-end and server-side concerns.

Reply:
@bbangert @jaredlockhart Since the Django app is `app/experimenter/`, would moving a React application into `app/nimbus-ui/` (and building into `app/nimbus-ui/dist/`) make serving it more difficult or would it be easy to point it here?

I like the idea of more FE and BE separation so I'd be fine with that if it doesn't complicate much. Though, one could argue the UI sits on top of `experimenter` and since the legacy UI will be in `/experimenter`, maybe it makes sense to leave it there. 🤷

Reply:
Yeah, totally fair! Y’all have far more insight into what would make sense here 🙂

Reply:
@LZoog Yeah so it depends on how we intend to serve it.  Right now all the React parts in the legacy and nimbus UIs are hosted using the Django static assets system and the [whitenoise module](http://whitenoise.evans.io/en/stable/), so they have to live inside the Django tree `app/experimenter/` so Django can find them.  If we want to host them separately, like directly with nginx, they could live outside the tree, but that would involve changing the way the production deployment is configured.  Either is doable, just needs some figuring out.

Reply:
This conversation is continued here: https://github.com/mozilla/experimenter/pull/3392/files#r485158559